socorro/processor/mozilla_transform_rules.py

```diff
@@ -816,3 +816,79 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
         except KeyError:
             return False
         return True
+
+
+#==============================================================================
+class BetaVersionRule(Rule):
+    required_config = Namespace()
+    required_config.add_option(
+        'database_class',
+        doc="the class of the database",
+        default=
+        'socorro.external.postgresql.connection_context.ConnectionContext',
+        from_string_converter=str_to_python_object,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.add_option(
+        'transaction_executor_class',
+        default="socorro.database.transaction_executor."
+                "TransactionExecutorWithInfiniteBackoff",
+        doc='a class that will manage transactions',
+        from_string_converter=str_to_python_object,
+        reference_value_from='resource.postgresql',
+    )
+
+    #--------------------------------------------------------------------------
+    def __init__(self, config):
+        super(BetaVersionRule, self).__init__(config)
+        database = config.database_class(config)
+        transaction = config.transaction_executor_class(
+            config,
+            database,
+        )
+        sql = """
+            SELECT
+                pv.product_name,
+                pv.release_version,
+                pv.build_type,
+                pvb.build_id,
+                pv.version_string
+            FROM product_versions pv
+                LEFT JOIN product_version_builds pvb ON
+                    (pv.product_version_id = pvb.product_version_id)
+        """
+        results = transaction(
+            execute_query_fetchall,
+            sql
+        )
```

Comment:
how frequently does this table change?  This class is instantiated once during processor startup so the content of this lookup table will persist unchanged until the SIGHUP or restart of the processor.